packages/snaps-execution-environments/src/common/endowments/abortController.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Creates wrapped AbortController endowment hardened by SES.
+ *
+ * @returns An object with the attenuated `AbortController` function.
+ */
+const createAbortController = () => {
+  /**
+   * Wrapped and attenuated AbortController.
+   */
+  const _AbortController = class implements AbortController {
+    #abortController: AbortController;
+
+    readonly signal: AbortSignal;
+
+    #signalHandler: ProxyHandler<object>;
+
+    constructor() {
+      this.#signalHandler = {
```

Comment:
There's no need for this function to exist in more than one copy, at least as-is. 
Are you planning on making it dependent on the scope of its closure?

Reply:
As previously discussed we can separate and track this work within another ticket and PR.
https://github.com/MetaMask/snaps-monorepo/issues/1132
I will resolve this thread here now because this is now removed from this PR.